yarn-project/p2p/src/client/p2p_client.ts

```diff
@@ -674,7 +674,7 @@ export class P2PClient<T extends P2PClientType = P2PClientType.Full>
       // every tx that's been generated against a block that has now been pruned is no longer valid
       if (tx.data.constants.historicalHeader.globalVariables.blockNumber.toNumber() > latestBlock) {
         const txHash = await tx.getTxHash();
-        txsToDelete.add(txHash);
+        txsToDelete.set(txHash.toString(), txHash);
```

Comment:
not necessary but u may also want to update the commented out code below

Reply:
I can't see any commented out code that uses this Set.